### PR TITLE
Added support for user to add their own docker compose fileName using -f flag

### PR DESCRIPTION
**Pull request by Aryakoste**
Closes: #1986 

Added -f flag for user to define their own docker compose file.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by Aryakoste** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Aryakoste** [comment]
recheck

File: pkg/core/app/app.go [hunk 1]
```diff
@@ -70,6 +71,16 @@ func (a *App) Setup(_ context.Context) error {
 	if utils.IsDockerKind(a.kind) && isDetachMode(a.logger, a.cmd, a.kind) {
 		return fmt.Errorf("application could not be started in detached mode")
 	}
+	var composeFilePath string
```
Comment (gouravkrosx) [inline pkg/core/app/app.go:74]: This is not the right way to do it, we will not parse the docker compose file path using a flag. Rather we will parse the appCmd and then get the filename.

**Comment by gouravkrosx** [comment]
@Aryakoste Please address the comment.

**Comment by Aryakoste** [comment]
Yeah will make the changes by EOD. And will let you know.

**Comment by Aryakoste** [comment]
@gouravkrosx Changes done. Check and let me know.

File: pkg/core/app/app.go [hunk 2]
```diff
@@ -65,11 +66,32 @@ type Options struct {
 	DockerNetwork string
 }
 
+func findComposeFilePathFromCmdArgs(args []string) string {
```
Comment (gouravkrosx) [inline pkg/core/app/app.go:69]: `findComposeFile` should handle all the logic of this, no need to make a separate function.

File: pkg/core/app/app.go [hunk 3]
```diff
@@ -65,11 +66,32 @@ type Options struct {
...
+	return ""
+}
+
 func (a *App) Setup(_ context.Context) error {
 
 	if utils.IsDockerKind(a.kind) && isDetachMode(a.logger, a.cmd, a.kind) {
 		return fmt.Errorf("application could not be started in detached mode")
 	}
+	var composeFilePath string
+	var cmdArgs []string
```
Comment (gouravkrosx) [inline pkg/core/app/app.go:84]: All logic related to the compose file should be in `SetupCompose` function only.

**Comment by Aryakoste** [comment]
@gouravkrosx Changes done.

File: pkg/core/app/app.go (hunk 2 above)
Comment (gouravkrosx) [reply pkg/core/app/app.go:69]: Please remove the older function

**Comment by gouravkrosx** [comment]
@Aryakoste Please fix the lint and DCO

File: pkg/core/app/util.go [hunk 4]
```diff
@@ -16,7 +16,14 @@ import (
 	"go.uber.org/zap"
 )
 
-func findComposeFile() string {
+func findComposeFile(cmdArgs []string) string {
```
Comment (gouravkrosx) [inline pkg/core/app/util.go:19]: it would be better if we pass the command rather than cmdArgs in the function param.

File: pkg/core/app/app.go [hunk 5]
```diff
@@ -130,11 +131,17 @@ func (a *App) SetupCompose() error {
...
+	cmdArgs = strings.Fields(a.cmd)
+
+	var path string
+	path = findComposeFile(cmdArgs)
 	if path == "" {
 		return errors.New("can't find the docker compose file of user. Are you in the right directory? ")
 	}
-	// kdocker-compose.yaml file will be run instead of the user docker-compose.yaml file acc to below cases
+
```
Comment (gouravkrosx) [inline pkg/core/app/app.go:144]: Also, put a debug log here that what path did we get.

**Comment by gouravkrosx** [comment]
@Aryakoste Please address the latest comments. I've tried your PR locally, it is working fine.

**Comment by Aryakoste** [comment]
I will make this changes till EOD.

**Comment by Aryakoste** [comment]
i tried to signoff all the commits and above thing happened is it ok.

**Comment by Aryakoste** [comment]
@gouravkrosx Changes done.

File: pkg/core/app/app.go [hunk 6]
```diff
@@ -130,11 +130,15 @@ func (a *App) SetupCompose() error {
...
-	path := findComposeFile()
+	// kdocker-compose.yaml file will be run instead of the user docker-compose.yaml file acc to below cases
+
+	path := findComposeFile(a.cmd)
 	if path == "" {
 		return errors.New("can't find the docker compose file of user. Are you in the right directory? ")
 	}
-	// kdocker-compose.yaml file will be run instead of the user docker-compose.yaml file acc to below cases
+
+	a.logger.Info(fmt.Sprintf("Obtained docker compose file path: %s", path))
```
Comment (gouravkrosx) [inline pkg/core/app/app.go:140]: it should be found rather than obtained

**Comment by gouravkrosx** [comment]
@Aryakoste Please fix the go lint and also change the log from obtained to found.

**Comment by Aryakoste** [comment]
@gouravkrosx Changes done.

**Review by gouravkrosx — APPROVED** [review]
LGTM!

**Pull request merged**